src/test/java/org/folio/auth/authtokenmodule/AuthTokenTest.java

```diff
@@ -654,286 +517,448 @@ public void testHttpEndpoints() throws JOSEException, ParseException {
+
+      logger.info("Test with extra permissions");
+      given()
+          .header("X-Okapi-Tenant", tenant)
+          .header("X-Okapi-Request-Id", "1234")
+          .header("X-Okapi-Token", moduleToken)
+          .header("X-Okapi-Url", "http://localhost:" + mockPort)
+          .header("X-Okapi-Permissions-Desired", "extra.*bar")
+          .get("/bar")
+          .then()
+          .statusCode(202)
+          .header("X-Okapi-Permissions", "[\"extra.foobar\"]");
+
+      logger.info("Test with extra permissions cached");
+      given()
+          .header("X-Okapi-Tenant", tenant)
+          .header("X-Okapi-Request-Id", "1234")
+          .header("X-Okapi-Token", moduleToken)
+          .header("X-Okapi-Url", "http://localhost:" + mockPort)
+          .header("X-Okapi-Permissions-Desired", "extra.*bar")
+          .get("/bar")
+          .then()
+          .statusCode(202)
+          .header("X-Okapi-Permissions", "[\"extra.foobar\"]");
+
+      logger.info("Test with extra permissions - timeout = 0");
+      ModulePermissionsSource.setCacheTimeout(0);
+      given()
+          .header("X-Okapi-Tenant", tenant)
+          .header("X-Okapi-Request-Id", "1234")
+          .header("X-Okapi-Token", moduleToken)
+          .header("X-Okapi-Url", "http://localhost:" + mockPort)
+          .header("X-Okapi-Permissions-Desired", "extra.*bar")
+          .get("/bar")
+          .then()
+          .statusCode(202)
+          .header("X-Okapi-Permissions", "[\"extra.foobar\"]");
+      ModulePermissionsSource.setCacheTimeout(60);
+
+      logger.info("Test with wildcard permission - zap cache");
+      given()
+          .header("Authtoken-Refresh-Cache", "true")
+          .header("X-Okapi-Tenant", tenant)
+          .header("X-Okapi-Request-Id", "1234")
+          .header("X-Okapi-Token", accessToken)
+          .header("X-Okapi-Url", "http://localhost:" + freePort)
+          .header("X-Okapi-Permissions-Desired", "extra.*bar")
+          .get("/bar")
+          .then()
+          .statusCode(400);
+    }
 
-    String headers = r.getHeader("X-Okapi-Permissions");
-    assertTrue(headers.contains(PermsMock.SYS_PERM_SUB_01));
-    assertTrue(headers.contains(PermsMock.SYS_PERM_SUB_02));
-  }
+    @Test
+    public void testExpandSystemPermission() {
+      Response r = given()
+          .header("X-Okapi-Tenant", tenant)
+          .header("X-Okapi-Request-Id", "1234")
+          .header("X-Okapi-Token", tokenSystemPermission)
+          .header("X-Okapi-Url", "http://localhost:" + mockPort)
+          .header("X-Okapi-Permissions-Required",
+              PermsMock.SYS_PERM_SUB_01 + "," + PermsMock.SYS_PERM_SUB_02)
+          .get("/testsysperm")
+          .then()
+          .statusCode(202)
+          .extract().response();
+
+      String headers = r.getHeader("X-Okapi-Permissions");
+      assertTrue(headers.contains(PermsMock.SYS_PERM_SUB_01));
+      assertTrue(headers.contains(PermsMock.SYS_PERM_SUB_02));
+    }
```

Comment:
Code coverage of /src/main/java decreases from 88,9 % to 88,2 %.
![2022-03-31-AuthTokenTest-a8311fe](https://user-images.githubusercontent.com/533612/161025427-3fb105ab-b6fd-4656-a053-58b1fd0be59d.png)
![2022-03-31-AuthTokenTest-ea9579d](https://user-images.githubusercontent.com/533612/161025452-294381c2-3cbf-4a0c-883f-686833ecefdb.png)

Reply:
Thanks Julian for thinking to run the coverage report. I had indeed missed one test case, which is now called `testSigningRequestGoodAccessTokenGoodPayload`. After adding this case the coverage report from Idea looks a bit better (using Idea and the JaCoCo coverage runner):

On master:
![image](https://user-images.githubusercontent.com/104455/161116620-9cdb281e-2200-4105-a1b5-1d0d96bd186f.png)

After adding missing case:
![image](https://user-images.githubusercontent.com/104455/161116652-405b1307-4256-4b74-90cb-015b81c17a15.png)

Here's what it looked like before the change:

![image](https://user-images.githubusercontent.com/104455/161116908-9953f384-a43e-4d55-820b-931ef3ce7f22.png)

I have gone through now a second time and checked that each case in both versions is there. How does it look in Eclipse?

Reply:
Well done, code coverage of /src/main/java is now exactly the same as before in Eclipse.